src/lua/datetime.lua

```diff
@@ -864,6 +878,73 @@ local function datetime_interval_add(lhs, rhs)
     end
 end
 
+--[[
+    Parse partial ISO-8601 date string
+
+    Accepetd formats are:
+
+    Basic      Extended
+    20121224   2012-12-24   Calendar date   (ISO 8601)
+    2012359    2012-359     Ordinal date    (ISO 8601)
+    2012W521   2012-W52-1   Week date       (ISO 8601)
+    2012Q485   2012-Q4-85   Quarter date
+
+    Returns pair of constructed datetime object, and length of string
+    which has been accepted by parser.
+]]
+local function datetime_parse_date(str)
+    check_str("datetime.parse_date()")
+    local dt = ffi.new('dt_t[1]')
+    local len = tonumber(builtin.tnt_dt_parse_iso_date(str, #str, dt))
+    if len == 0 then
+        error(('invalid date format %s'):format(str), 2)
+    end
+    return datetime_new_dt(dt[0]), len
+end
+
+--[[
+    datetime parse function for strings in extended iso-8601 format
+    assumes to deal with date T time time_zone at once
+       date [T] time [ ] time_zone
+    Returns constructed datetime object.
+]]
+local function datetime_parse_full(str, tzoffset)
+    check_str(str, "datetime.parse()")
+    local date = ffi.new(datetime_t)
+    local rc = builtin.tnt_datetime_parse_full(date, str, #str, tzoffset)
+    if rc == false then
+        error(("Could not parse '%s'"):format(str))
+    end
+    return date
+end
+
+local function datetime_parse_from(str, obj)
+    check_str(str, "datetime.parse()")
+    local fmt = ''
```

Comment:
According to commit message 'format' param is optional: "`datetime.parse` function expect 1 required argument - which is input string, and set of optional parameters passed as table in 2nd argument.", in spec: "без указания формата - эвристический формат (то что есть сейчас, понимающее разные форматы “похожие” на 8601 с разного рода эвристикой: с пробелами/без и т.п.)"

But:
```
tarantool> date.parse('12/31/2020')
---
- error: 'builtin/datetime.lua:918: Could not parse ''12/31/2020'''
```

Reply:
This ordering is not expected by any of ISO-8601 formats or of their extensions we have implemented via `c-dt`.

Though, actually we would need to have it parsed (eventually) for SQL `DATE`. if session settings would be `dmy` instead of default `ymd`. But I don't expect it to be available in the 1st iteration.